cellpack/autopack/loaders/recipe_loader.py

```diff
@@ -124,25 +127,29 @@ def _sanitize_format_version(recipe_data):
             format_version = recipe_data["format_version"]
         return format_version
 
-    def _migrate_version(self, recipe):
-        new_recipe = {}
+    def _migrate_version(self, old_recipe):
+        converted = False
+        if old_recipe["format_version"] == "1.0":
+            converted = True
+            new_recipe = convert_v1_to_v2(old_recipe)
+            old_recipe = copy.deepcopy(new_recipe)
 
-        if recipe["format_version"] == "1.0":
-            new_recipe["version"] = recipe["recipe"]["version"]
-            new_recipe["format_version"] = self.current_version
-            new_recipe["name"] = recipe["recipe"]["name"]
-            new_recipe["bounding_box"] = recipe["options"]["boundingBox"]
-            (
-                new_recipe["objects"],
-                new_recipe["composition"],
-            ) = convert(recipe)
+        if old_recipe["format_version"] == "2.0":
+            new_recipe = copy.deepcopy(old_recipe)
+            converted = True
+
+            new_recipe = convert_v2_to_v2_1(old_recipe)
+
+        if converted:
             if self.save_converted_recipe:
                 self._save_converted_recipe(new_recipe)
+
+            return new_recipe
+
         else:
             raise ValueError(
-                f"{recipe['format_version']} is not a format version we support"
+                f"{old_recipe['format_version']} is not a format version we support"
             )
```

Comment:
I see how we are handling version migration here! The refactor of separate conversion functions for different versions and the `converted` flag has really improved the readability of the code.